x-pack/plugins/actions/server/lib/task_runner_factory.ts

```diff
@@ -89,10 +91,22 @@ export class TaskRunnerFactory {
     const taskInfo = {
       scheduled: taskInstance.runAt,
       attempts: taskInstance.attempts,
+      requeueInvalidTask: taskInstance.requeueInvalidTask,
     };
     const actionExecutionId = uuidv4();
     const actionTaskExecutorParams = taskInstance.params as ActionTaskExecutorParams;
 
+    const shouldSkip = ({ status, message }: ActionTypeExecutorResult<unknown>) => {
```

Comment:
It seems to me that this logic should only be handled by the task manager task runner. I don't think task types should manage this logic.

Reply:
without `beforeRun`, i had to add it here...